core/include/vecmem/containers/impl/jagged_vector_buffer.ipp

```diff
@@ -102,9 +106,13 @@ jagged_vector_buffer<TYPE>::jagged_vector_buffer(
 }
 
 template <typename TYPE>
+template <typename SIZE_TYPE,
+          std::enable_if_t<std::is_integral<SIZE_TYPE>::value &&
+                               std::is_unsigned<SIZE_TYPE>::value,
```

Comment:
The second one implies the first so there was no need for both checks though

Reply:
`std::is_unsigned<bool>::value == true`

Reply:
Just for reference, `std::is_integral<bool>::value` is also true. :wink:

Reply:
Yeah, I realised after writing this down... :frowning: Okay, so `std::is_integral` is indeed quite useless at the moment...

Reply:
Just to be clear, I do actually like the `std::is_integral` being there. `std::is_unsigned` is not defined as the type being a member of a predefined set of unsigned types, but rather it is defined as `std::is_unsigned<T>` iff. T(-1) > T(1).

It's rather easy to come up with a pathological unsigned floating point type that the standard library would classify as unsigned! So the integral check does have a use. :slightly_smiling_face: